### ISSUE TITLE
[FR] Add ability to utilize existing sagemaker model for batch transform job

### ISSUE THREAD

**Issue by clarkh-ncino** [mlflow/mlflow#9632]
### Willingness to contribute

Yes. I would be willing to contribute this feature with guidance from the MLflow community.

### Proposal Summary

Add the ability to use an existing sagemaker model for batch transform jobs rather than creating a new model for each job.

### Motivation

> #### What is the use case for this feature?
When using [deploy_transform_job()](https://mlflow.org/docs/latest/python_api/mlflow.sagemaker.html#mlflow.sagemaker.deploy_transform_job), a new model is created and used for each batch inference job. There is no reuse of existing models.
> #### Why is this use case valuable to support for MLflow users in general?
This is valuable because without the ability to utilize existing models you lose benefits of model versioning like:
Maintaining multiple versions of a model in production
Tracking model lineage and metrics per version
> #### Why is this use case valuable to support for your project(s) or organization?
We rely heavily on MLFlow for many things in our AI/ML workflow. Deployments are one of the biggest areas and this would keep us aligned with our strategy to utilize MLFlow exclusively rather than mixing boto3 and MLFlow together. 
> #### Why is it currently difficult to achieve this use case?
This functionality is not supported yet out of the box with MLFlow.

### Details

Add a new optional parameter `model_name` to [deploy_transform_job()](https://mlflow.org/docs/latest/python_api/mlflow.sagemaker.html#mlflow.sagemaker.deploy_transform_job) to check for an existing model in AWS sagemaker before creating a new model, if model exists use that model, if not create a new model and execute the transform job.
Add a check for model_name param, if it is set, use that for the model name, if not create the model_name based off the transform job ([which is the current logic](https://github.com/mlflow/mlflow/blob/6de2829a28898001aabcc681750d03c5cd291399/mlflow/sagemaker/__init__.py#L757)). 

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [X] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by BenWilson2** [comment]
This sounds like a good idea. Please feel free to ping us with any questions while working on the PR! 
One quick question about this before you get started: 
If a model is registered with a particular name in sagemaker (i.e., "my_transform_job_1") and you have the registered model name in MLflow for a completely different job as that, are you planning on validating the consistency of the model name reference to what is currently in Sagemaker? Can we generate a unique identifier id that is appended to the name (or generated) that will guarantee that the end user's intentions are respected?
Would it make sense to use a reserved tag for this (from your other FR) that can first be fetched from sagemaker to see if the model_uri that originated the transform request already exists in sagemaker? 

Just some things to think about with respect to ensuring that we're not breaking any intended functionality. A component to consider with this as well is handling legacy models that have been pushed that don't support that reserved tagging (or any other component other than name, which might not be the best disambiguation methodology) such that reuse is only possible if using the added mechanism for correctness guarantee.

**Comment by clarkh-ncino** [comment]
> This sounds like a good idea. Please feel free to ping us with any questions while working on the PR! One quick question about this before you get started: If a model is registered with a particular name in sagemaker (i.e., "my_transform_job_1") and you have the registered model name in MLflow for a completely different job as that, are you planning on validating the consistency of the model name reference to what is currently in Sagemaker? Can we generate a unique identifier id that is appended to the name (or generated) that will guarantee that the end user's intentions are respected? Would it make sense to use a reserved tag for this (from your other FR) that can first be fetched from sagemaker to see if the model_uri that originated the transform request already exists in sagemaker?
> 
> Just some things to think about with respect to ensuring that we're not breaking any intended functionality. A component to consider with this as well is handling legacy models that have been pushed that don't support that reserved tagging (or any other component other than name, which might not be the best disambiguation methodology) such that reuse is only possible if using the added mechanism for correctness guarantee.

Yep, I think appending a UID to a model name when created is always a good idea. Looks like that is happening for most sagemaker infrastructure already, including the model created while deploying a transform job. https://github.com/mlflow/mlflow/blob/c93497f235933005c7ed39956b469d936652f9eb/mlflow/sagemaker/__init__.py#L1337-L1346

**Comment by BenWilson2** [comment]
Perfect! I'll look forward to your PR :)
